source/text/Format.cpp

```diff
@@ -27,7 +27,9 @@ this program. If not, see <https://www.gnu.org/licenses/>.
 using namespace std;
 
 namespace {
-	const int64_t K = 1000;
+	// The greatest number displayed without switching to scientific notation
+	constexpr int64_t SCIENTIFIC_THRESHOLD = 1000000000000000ll;
```

Comment:
if 1e15 doesn't work, at least do this:
```suggestion
	constexpr int64_t SCIENTIFIC_THRESHOLD = 1'000'000'000'000'000ll;
```